synapse/storage/schema/delta/55/room_retention.sql

```diff
@@ -19,9 +19,12 @@
 -- If a room doesn't have a retention policy state event in its state, both max_lifetime
 -- and min_lifetime are NULL.
 CREATE TABLE IF NOT EXISTS room_retention(
-    room_id TEXT PRIMARY KEY,
+    room_id TEXT NOT NULL,
+    event_id TEXT,
```

Comment:
this should be NOT NULL too?

Reply:
No, there's a composite primary key on `(room_id, event_id)` preventing either from being NULL.

Reply:
ok, so why make `room_id` `NOT NULL` ? at the moment it looks like one is nullable and the other isn't.

Reply:
FTR this was a remain from a previous experimentation that I forgot to remove, and it got fixed in later commits.